tests/test_log_command.rs

```diff
@@ -989,3 +989,108 @@ fn test_graph_styles() {
     o
     "###);
 }
+
+#[test]
+fn test_log_word_wrap() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let render = |args: &[&str], columns: u32, word_wrap: bool| {
+        let mut args = args.to_vec();
+        if word_wrap {
+            args.push("--config-toml=ui.log-word-wrap=true");
+        }
+        let assert = test_env
+            .jj_cmd(&repo_path, &args)
+            .env("COLUMNS", columns.to_string())
+            .assert()
+            .success()
+            .stderr("");
+        get_stdout_string(&assert)
+    };
+
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "main branch 1"]);
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m", "main branch 2"]);
+    test_env.jj_cmd_success(&repo_path, &["new", "-m", "side"]);
+    test_env.jj_cmd_success(&repo_path, &["new", "-m", "merge", "@--", "@"]);
+
+    // ui.log-word-wrap option applies to both graph/no-graph outputs
+    insta::assert_snapshot!(render(&["log", "-r@"], 40, false), @r###"
+    @  mzvwutvlkqwt test.user@example.com 2001-02-03 04:05:11.000 +07:00 68518a7e6c9e
+    │  (empty) merge
+    ~
+    "###);
+    insta::assert_snapshot!(render(&["log", "-r@"], 40, true), @r###"
+    @  mzvwutvlkqwt test.user@example.com
+    │  2001-02-03 04:05:11.000 +07:00
+    ~  68518a7e6c9e
+       (empty) merge
+    "###);
+    insta::assert_snapshot!(render(&["log", "--no-graph", "-r@"], 40, false), @r###"
+    mzvwutvlkqwt test.user@example.com 2001-02-03 04:05:11.000 +07:00 68518a7e6c9e
+    (empty) merge
+    "###);
+    insta::assert_snapshot!(render(&["log", "--no-graph", "-r@"], 40, true), @r###"
+    mzvwutvlkqwt test.user@example.com
+    2001-02-03 04:05:11.000 +07:00
+    68518a7e6c9e
+    (empty) merge
+    "###);
+
+    // Graph width should be subtracted from the term width
+    let template = r#""0 1 2 3 4 5 6 7 8 9""#;
+    insta::assert_snapshot!(render(&["log", "-T", template], 10, true), @r###"
+    @    0 1 2
+    ├─╮  3 4 5
+    │ │  6 7 8
+    │ │  9
+    o │  0 1 2
+    │ │  3 4 5
+    │ │  6 7 8
+    │ │  9
+    o │  0 1 2
+    ├─╯  3 4 5
+    │    6 7 8
+    │    9
+    o  0 1 2 3
+    │  4 5 6 7
+    │  8 9
+    o  0 1 2 3
+       4 5 6 7
+       8 9
+    "###);
+    insta::assert_snapshot!(
+        render(&["log", "-T", template, "--config-toml=ui.graph.style='legacy'"], 9, true),
+        @r###"
+    @   0 1 2
+    |\  3 4 5
+    | | 6 7 8
+    | | 9
+    o | 0 1 2
+    | | 3 4 5
+    | | 6 7 8
+    | | 9
+    o | 0 1 2
+    |/  3 4 5
+    |   6 7 8
+    |   9
+    o 0 1 2 3
+    | 4 5 6 7
+    | 8 9
+    o 0 1 2 3
+      4 5 6 7
+      8 9
+    "###);
+
+    // Shouldn't panic with COLUMNS=0
+    insta::assert_snapshot!(render(&["log", "-r@"], 0, true), @r###"
```

Comment:
Optional: I'd add the same test with COLUMNS=1 for documentation purposes, so that people know COLUMS=0 is not exceptional.

Reply:
Ok, added the one.

Just fyi, `COLUMNS=0` may be interpreted differently by shell. Here we execute the command directly, so both `COLUMNS=0` and `=1` should work.

Reply:
> Just fyi, COLUMNS=0 may be interpreted differently by shell. 

What do you mean? I haven't heard of anything like that. Do shells interpret `COLUMNS=0` as `COLUMNS=`, unsetting the variable?

Reply:
Yes, at least for my zsh configuration:
```
% COLUMNS=0 env | grep COLUMNS
COLUMNS=547
% env COLUMNS=0 env | grep COLUMNS
COLUMNS=0
```

I guess `COLUMNS=` made a shell variable exported, and invalid values like `0` would be ignored.